diff --git a/rc_init.c b/rc_init.c
index 0ab398a..778508c 100644
--- a/rc_init.c
+++ b/rc_init.c
@@ -179,7 +179,7 @@ static int  rc_slave_cfg(struct scsi_device *sdev);
 #else
 static int  rc_slave_cfg(struct scsi_device *sdev, struct queue_limits *qlimits);
 #endif
-int         rc_bios_params(struct scsi_device *sdev, struct block_device *bdev,
+int         rc_bios_params(struct scsi_device *sdev, struct gendisk *bdev,
 			   sector_t capacity, int geom[]);
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,36)
@@ -1919,7 +1919,7 @@ rc_info (struct Scsi_Host *host_ptr)
  */
 int
 rc_bios_params (struct scsi_device *sdev,
-		struct block_device *bdev,
+		struct gendisk *bdev,
 		sector_t capacity,
 		int geom[])
 {
